server/api/contest_handler.py

```diff
@@ -84,30 +95,34 @@ def get_contest(contest_id):
     except Exception:
         return jsonify("Contest does not exist")
 
-    #Load user info
-    try:
-        user = User.query.filter_by(id=contest.contest_creater).first()
-        if user == None: 
-            raise Exception
-    except Exception:
-        return jsonify("Contest owner not found")
-    
-    #Load submissions
-    allSubmissions = db.session.query(Submission.image_link, User.username).filter_by(contest_id=contest_id).join(User, User.id == Submission.submiter_id).all()
-    
-    formatedSubmissions = []
+    if request.method == 'GET':
+        #Load user info
+        try:
+            contest_creater_user = User.query.filter_by(id=contest.contest_creater).first()
+            if contest_creater_user == None: 
+                raise Exception
+        except Exception as e:
+            return jsonify("Contest owner not found")
 
-    for pair in allSubmissions:
-        imgLink, username = pair
-        formatedSubmissions.append({
-            "img": imgLink,
-            "creater": username
-        })
+        if contest_creater_user == current_user:
```

Comment:
what happens if you're not the `contest_creater_user`? it should be that you only see your own submissions right?

also its better to compare the IDs rather than the entire object :)